PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h

```diff
@@ -4257,12 +4276,37 @@ void CheckCurrentRunNumber(T1 const& collision, T2 const&)
   // a) Support for Run 3 real data:
   if constexpr (rs == eRec) {
 
-    auto bc = collision.template foundBC_as<T2>(); // TBI 20241126 I have the same code snippet at 2 other locations, I shall unify at some point
+    // **) Check run number:
+    auto bc = collision.template foundBC_as<T2>(); // I have the same code snippet at other places, keep in sync.
     if (!tc.fRunNumber.EqualTo(Form("%d", bc.runNumber()))) {
       LOGF(error, "\033[1;33m%s Run number changed within process(). This most likely indicates that a given masterjob is processing 2 or more different runs in one go.\033[0m", __FUNCTION__);
       LOGF(fatal, "tc.fRunNumber = %s, bc.runNumber() = %d", tc.fRunNumber.Data(), bc.runNumber());
     }
 
+    // **) Check SoR, EoR, and run duration:
+    auto runDuration = ccdb->getRunDuration(bc.runNumber());    // this is total run duration, not the current one (see below)
+    int64_t startOfRun = std::floor(runDuration.first * 0.001); // in seconds since Unix epoch
+    int64_t endOfRun = std::ceil(runDuration.second * 0.001);   // in seconds since Unix epoch
+    int64_t durationInSec = endOfRun - startOfRun;              // yes, this is now run duration in seconds
+
+    // **) Insanity check on SoR:
+    if (!(tc.fRunTime[eStartOfRun] == startOfRun)) {
+      LOGF(error, "\033[1;33m%s tc.fRunTime[eStartOfRun] changed within process(). This most likely indicates that a given masterjob is processing 2 or more different runs in one go.\033[0m", __FUNCTION__);
+      LOGF(fatal, "tc.fRunTime[eStartOfRun] = %d, startOfRun = %d", tc.fRunTime[eStartOfRun], startOfRun);
+    }
+
+    // **) Insanity check on EoR:
+    if (!(tc.fRunTime[eEndOfRun] == endOfRun)) {
+      LOGF(error, "\033[1;33m%s tc.fRunTime[eEndOfRun] changed within process(). This most likely indicates that a given masterjob is processing 2 or more different runs in one go.\033[0m", __FUNCTION__);
+      LOGF(fatal, "tc.fRunTime[eEndOfRun] = %d, endOfRun = %d", tc.fRunTime[eEndOfRun], endOfRun);
+    }
+
+    // **) Insanity check on run duration:
+    if (!(tc.fRunTime[eDurationInSec] == durationInSec)) {
+      LOGF(error, "\033[1;33m%s tc.fRunTime[eDurationInSec] changed within process(). This most likely indicates that a given masterjob is processing 2 or more different runs in one go.\033[0m", __FUNCTION__);
+      LOGF(fatal, "tc.fRunTime[eDurationInSec] = %d, durationInSec = %d", tc.fRunTime[eDurationInSec], durationInSec);
+    }
```

Comment:
If this is being done per BC is useless because the object from the CCDB is always the same it only changes when the run number changes and this usually does not happen at the grid nodes
Of course, locally it can be enforced

Reply:
I do NOT use this check regularly: "  // Used only during validation."

It's a protection that I do not process in one masterjob data from different runs. Once that is checked, the call to this function is switched off. 

In practice, this function is never called, unless weird things start to happen...